src/Core/WinSWCore/ServiceDescriptor.cs

```diff
@@ -95,6 +100,40 @@ public ServiceDescriptor(XmlDocument dom)
             this.dom = dom;
         }
 
+        private void ValidateAndLoadXmlSchema()
+        {
+            XmlReaderSettings settings = new XmlReaderSettings();
+
+            try
+            {
+                settings.Schemas.Add(null, BasePath + ".xsd");
+            }
+            catch (FileNotFoundException)
+            {
+                throw new FileNotFoundException("XSD file not found");
+            }
+
+            settings.ValidationType = ValidationType.Schema;
+            settings.ValidationEventHandler += new ValidationEventHandler(XmlValidationEventHandler);
+            
+            using(XmlReader reader = XmlReader.Create(BasePath + ".xml"))
+            {
+                dom.Load(reader);
+            }
+        }
+
+        private void XmlValidationEventHandler(object sender, ValidationEventArgs e)
+        {
+            if (e.Severity == XmlSeverityType.Warning)
+            {
+                throw new XmlException("[Warning] XML validation - " + e.Message);

```

Comment:
I'm wondering if we need to throw for warnings at all. A warning is typically issued when there is no schema to validate a particular element, e.g. after we add a new setting but do not update the XSD.

Reply:
Yes. Its better to move on with just Errors.